src/entities/allocator.ts

```diff
@@ -79,6 +80,9 @@ export function createOrLoadAllocator(allocatorAddress: Address, vaultAddress: A
 
   if (vaultAllocator === null) {
     vaultAllocator = new Allocator(vaultAllocatorAddress)
+    if (ethereum.hasCode(allocatorAddress).inner) {
```

Comment:
Whether it’s a contract or an EOA matters only during periodic distributions. For example, currently, we don’t check whether an osTokenHolder is a contract. Every time we add a new entity that can participate in periodic distributions, we should perform the same check. Therefore, we should check and store whether each user address is an EOA or a contract during distribution updates, and do so for every unseen address. By centralizing this, it reduces the potential for errors and makes it easier to maintain.